packages/extensions/venia-sample-payments-checkmo/src/talons/useCheckmo.js

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useEffect } from 'react';
+import { useQuery, useMutation } from '@apollo/client';
+import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+
+import DEFAULT_OPERATIONS from './checkmo.gql';
+
+/**
+ * Talon to handle checkmo payment.
+ *
+ * @param {Boolean} props.shouldSubmit boolean value which represents if a payment nonce request has been submitted
+ * @param {Function} props.onPaymentSuccess callback to invoke when the a payment nonce has been generated
+ * @param {Function} props.onPaymentError callback to invoke when component throws an error
+ * @param {Function} props.resetShouldSubmit callback to reset the shouldSubmit flag
+ * @param {DocumentNode} props.operations.getCheckmoConfigQuery query to fetch config from backend
+ * @param {DocumentNode} props.operations.setPaymentMethodOnCartMutation mutation to set checkmo as payment
+ *
+ * @returns {
+ *  payableTo: String,
+ *  mailingAddress: String,
+ *  onBillingAddressChangedError: Function,
+ *  onBillingAddressChangedSuccess: Function
+ * }
+ */
+export const useCheckmo = props => {
+    const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
+
+    const {
+        getCheckmoConfigQuery,
+        setPaymentMethodOnCartMutation
+    } = operations;
+
+    const [{ cartId }] = useCartContext();
+    const { data } = useQuery(getCheckmoConfigQuery);
+
+    const { resetShouldSubmit, onPaymentSuccess, onPaymentError } = props;
+
+    const [
+        updatePaymentMethod,
+        {
+            error: paymentMethodMutationError,
+            called: paymentMethodMutationCalled,
+            loading: paymentMethodMutationLoading
+        }
+    ] = useMutation(setPaymentMethodOnCartMutation, {
+        skip: !cartId,
```

Comment:
I don't know if you need a `skip`. You also should be passing the variables to the mutation call in the callback, not when you call the hook here.

```
updatePaymentMethod({
  variables: { cartId }
});
```